### DISCUSSION TITLE
Add widgets

### DISCUSSION THREAD

**Discussion by amithiel** [mullvad/mullvadvpn-app#8946] [Feature requests & Ideas]
Hi!

For someone using mullvad since the old days, many years ago, i watch you guys grow. It always have been that reliable VPN. Anyway, may i add some suggestions to your android client?

**Widgets:**

Lets suppose you text someone on a messaging client, and you're waiting for the reply. The person is taking too long, and you start wondering. You pull the notification banner, mullvad says connected. In reallity, its not connected. Soon you realize you need to change server or reconnect and everything starts working. A simple glance at your homescreen widget, and you can see the connection statistics. No data being downloaded/uploaded? sure, something going on with that server/connection, just change server and don't waste time. A dark and simplistic widget please, i may add.

**Server load:**

This is a big one. You select the country, now there are a bunch of servers...which one you chose? i'd rather chose the one with the least load/connect users. I would be nice to know before connecting.

Other than those, i think this vpn is perfect and getting better and better...

Note: sorry for grammar mistakes, english is not my main language

**Comment by faern** [comment]
### Widget

That could be useful indeed. Thanks for the suggestion. We'll think about it. However, the app should already as is detect if a tunnel is broken and reconnect to a new server automatically. But the timeout is rather long, so it could take a minute or two before it does reconnect after the tunnel actually becomes broken.

### Server load

Server load is just a false sense of performance. We don't expose it because it has very little to do with the actual tunnel performance you will see when connected to it. We have plenty of spare bandwidth and CPU on all our servers. See this FAQ: https://mullvad.net/en/help/tag/connectivity/#69

The only thing you can do really is try a server and compare it with another. We will soon expose which provider the server is hosted at in the app, then it will be easier to try different servers from different providers in the same location. That's usually what makes the biggest difference since it can affect what routing/peering you will have available between yourself and the server.

**Comment by faern** [comment]
It's also not very practical to have two things in the same issue. Because closing it means closing both things. The server load won't happen, but it might make sense to track the feature request on the widget. I'll rename the issue and from now on it will track having an Android widget.
